docs/design-documents/tools/cmake.md

```diff
@@ -1,101 +1,118 @@
+## Building an Application
 
-By default, baremetal would be selected - requires set to hal, platform, drivers and cmsis. If an app needs anything else, would use `requires` in the config to include - BLE/networking/etc.
+`mbed-tools` is the next generation of command line tooling for Mbed OS. `mbed-tools` replaces the `mbed-cli` and the Python modules in the `mbed-os/tools` directory.
+
+`mbed-tools` has consolidated all of the required modules to build Mbed OS, along with the command line interface, into a single Python package which can be installed using standard python packaging tools.
 
 A user create own CMake file to configure an application, also with `mbed_app.json` configuration file. The building of an app would look like:
 
 1. Parse the arguments provided to build command
-2. Parse the application configuration
-3. Get the target configuration
-4. Get the Mbed OS configuration (select what modules we need and get their config, paths, etc)
-5. Create the toolchain.cmake file
-6. Inject all previous steps to the Mbed OS Core CMake
-7. Build an application
+1. Parse the application configuration
+1. Get the target configuration
+1. Get the Mbed OS configuration (select what modules we need and get their config, paths, etc)
+1. Create .mbedbuild/mbed_config.cmake
+1. Build an application
+
+### Configuration
+
+The main purpose of `mbed-tools` is to parse the Mbed configuration system's JSON files (`mbed_lib.json`, `mbed_app.json` and `targets.json`). The tool outputs a single CMake configuration script, which is included by `app.cmake` and `mbed-os/CMakeLists.txt`.
+
+To generate the CMake config script (named `mbed_config.cmake`) the user can run the `configure` command:
+
+`mbedtools configure -t <toolchain> -m <target>`
+
+This will output `mbed_config.cmake` in a directory named `.mbedbuild` at the root of the program tree.
+
+`mbed_config.cmake` contains several variable definitions used to select the toolchain, core and profile CMake scripts to be used in the build system generation:
+* `MBED_TOOLCHAIN`
+* `MBED_TARGET`
+* `MBED_CPU_CORE`
+* `MBED_PROFILE`
+
+The tools also generate an `MBED_TARGET_LABELS` variable, containing the labels, components and feature definitions from `targets.json`, used to select the required Mbed OS components to be built.
+
+The macro definitions parsed from the Mbed OS configuration system are also included in `mbed_config.cmake`. The decision was made to remove `mbed_config.h`.
+
+### mbedignore 
+
+TBD
```

Comment:
```suggestion
With a build system that can understand dependencies between applications, features, components, and targets, `mbedignore` is no longer needed. CMake creates a list of exactly what's needed for an application build, rather than do what the legacy tools did and include everything by default, leaving the application to specify what is not needed. It's far more natural to express what you need rather than what you don't need, not to mention more future proof should some new feature appear in Mbed OS which your application would need to ignore.
```

Reply:
We should think about migration path for users , if there exists a way for us to facilitate migration the old code base - for example, can we find out there is mbedignore and warn users about this functionality is not part of Mbed OS 7 and point them to a migration guide? This should be addressed in mbed-tools (for CMake, mbedignore does not exists).